src/styles/index.scss

```diff
@@ -1,3 +1,54 @@
-body {
-  margin: 0;
+@import './abstract/constants';
+@import './abstract/placeholders';
+@import './base/base';
+
+.calendar {
+  display: flex;
+  padding: $calendar-indent;
+  flex-wrap: wrap;
+  align-items: center;
+  gap: $gap;
+
+  @extend %calendar-width;
+
+  @each $day, $index in $days-of-week {
+    &--start-day-#{$day} :first-child {
+      margin-left: ($day-size + $gap) * $index;
+    }
+  }
+
+  @for $length from $min-days through $max-days {
+    &--month-length-#{$length} {
+      @for $day from ($length) through $max-days {
+        .calendar__day:nth-child(#{$day + 1}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &__day {
+    font-size: 30px;
+    cursor: pointer;
+    position: relative;
+    width: $day-size;
+    height: $day-size;
+    border: 1px solid $day-border-color;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: $day-bgc-color;
+
+    @for $day from 1 through $max-days {
+      &:nth-child(#{$day})::after {
+        content: '#{$day}';
+      }
+    }
+
+    &:hover {
+      transform: translateY(-20px);
+      transition: 0.5s;
```

Comment:
The transition property should be moved to the original selector, not the hover state. This way, the transition will be smooth in both directions, not just when the mouse is hovering over the element. (Code Quality Issue)